File: gradle-java-server/src/main/java/com/github/genraven/gradlejavaserver/domain/Modifier.java
package com.github.genraven.gradlejavaserver.domain;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

@Data
public class Modifier {
    private Type type;
    private int ranks;

    @AllArgsConstructor
    @Getter
    public enum Type {
        INCREASE_ENCUMBRANCE_CAPACITY("Increase Encumbrance Capacity"),
        INCREASE_BASE_ENCUMBRANCE_CAPACITY("Increase Base Encumbrance Capacity"),
        INCREASE_WOUND_THRESHOLD("Increase Wound Threshold"),
        SUFFER_WOUNDS("Suffer Wounds"),
        SUFFER_WOUNDS_BY_TALENT("Suffer Wounds By Talent"),
        INCREASE_CRITICAL_INJURY("Increase Critical Injury"),
        DECREASE_CRITICAL_INJURY("Decrease Critical Injury"),
        RESET_CRITICAL_INJURY("Reset Critical Injury");

        @JsonValue
        private final String label;
    }
}

File: client/src/models/common/Modifier.ts
import {Option} from '../../components/common/InputSelectField';

export default interface Modifier {
    type: Type
    ranks: number
}

export enum Type {
    IncreaseEncumbranceCapacity = 'Increase Encumbrance Capacity',
    IncreaseBaseEncumbranceCapacity = 'Increase Base Encumbrance Capacity',
    MoveStoryPoint = 'Move Story Point',
    IncreaseSoak = 'Increase Soak',
    IncreaseWoundThreshold = 'Increase Wound Threshold',
    SufferWounds = 'Suffer Wounds',
    SufferWoundsByTalent = 'Suffer Wounds By Talent',
    SufferStrain = 'Suffer Strain',
    IncreaseCriticalInjury = 'Increase Critical Injury',
    DecreaseCriticalInjury = 'Decrease Critical Injury',
    ResetCriticalInjury = 'Reset Critical Injury',
    AddSetbackOnTargetedSocial = 'Add Setback on Targeted Social'
}

export const getModifierOptions = (): Option[] => {
    return Object.values(Type).map((value) => ({value}))
}